flink-core/src/main/java/org/apache/flink/configuration/TaskManagerOptions.java

```diff
@@ -83,12 +85,27 @@
 	public static final ConfigOption<String> HOST =
 		key("taskmanager.host")
 			.noDefaultValue()
-			.withDescription("The hostname of the network interface that the TaskManager binds to. By default, the" +
-				" TaskManager searches for network interfaces that can connect to the JobManager and other TaskManagers." +
-				" This option can be used to define a hostname if that strategy fails for some reason. Because" +
-				" different TaskManagers need different values for this option, it usually is specified in an" +
+			.withDescription("The address of the network interface that the TaskManager binds to." +
+				" This option can be used to define explicitly a binding address. Because" +
+				" different TaskManagers need different values for this option, usually it is specified in an" +
 				" additional non-shared TaskManager-specific config file.");
 
+	/**
+	 * The config parameter for automatically defining the TaskManager's binding address,
+	 * if {@link #HOST} configuration option is not set.
+	 */
+	public static final ConfigOption<String> HOST_BIND_POLICY =
+		key("taskmanager.host.bind-policy")
+		.defaultValue("hostname")
```

Comment:
This default is fine if we follow my other suggestion. If not, this should be `auto-detect-hostname`.

Reply:
I agree with Ufuk. My fear is that we might break existing setups with this change. In particular since we do this change so close to the feature freeze I'm feeling a bit uneasy. Thus, I would recommend to use `auto-detect-hostname` as default.